cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillRunnerService.java

```diff
@@ -311,13 +316,66 @@ public void resourceModified(T resource) {
             startupTaskCompletion.complete(null);
             // Cancel the scheduled termination
             terminationFuture.cancel(false);
+            // Start the job monitor if we haven't already done so.
+            // This is a workaround for CDAP-19134 - sometimes the watch doesn't get triggered on job completion.
+            if (!wrappedFuture.isDone()) {
+              Future<?> future = Executors.newSingleThreadScheduledExecutor(
+                  Threads.createDaemonThreadFactory(String.format("job-monitor-%s", runId)))
```

Comment:
So this can potentially starts a lot of new threads if there are many jobs running? Why can't we use the `monitorScheduler`?

Reply:
Yes, this can start a lot of threads if there are many jobs running.
I have updated the PR to use `monitorScheduler`.
After discussing with @albertshau decided to fix this issue by specifying [resource version](https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-watch) on the watch to ensure that we don't lose events. I have implemented this approach in https://github.com/cdapio/cdap/pull/14396.

Reply:
So does this change is still needed? It's not clear what problem was it and how we are addressing it.

Reply:
This change is not needed. This issue will be fixed in https://github.com/cdapio/cdap/pull/14396. 
Problem description is in the other PR - https://github.com/cdapio/cdap/pull/14396#issuecomment-1179223972